Add a CHANGELOG file

tedsuo opened the pull request:
In the same manner as the golang changelog: https://github.com/opentracing/opentracing-go/blob/master/CHANGELOG.md

pavolloffay commented on CHANGELOG.md:
```diff
@@ -0,0 +1,13 @@
+Changes by Version
+==================
```
Could be used the same formatting symbols as in other markdown files in this repository. e.g. `#` for headers, `*` for lists

tedsuo replied:
I was trying to be uniform by copying the style in the go CHANGELOG ;) : https://raw.githubusercontent.com/opentracing/opentracing-go/master/CHANGELOG.md\

We also use `.rst` instead of `.md` in some repos. 🤷‍♂️ 

I prefer `#` and `*` so I'll switch to that. Maybe we can standardize and change the others as part of the upcoming "language maintainer" initiative.

pavolloffay commented on CHANGELOG.md:
```diff
@@ -0,0 +1,13 @@
+==================
+
+v0.31.0 (2018-01-12)
+--------------------
```
Could we add one list item per added change? All the following would be in the one list item. Also, add a link to the PR, in some projects there is also a name of the commiter.

Two major changes are missing there:
* introduction of log fields
* removed `span.log(object)` methods

tedsuo replied:
I'm not sure what you mean by "All the following would be in the one list item." Do you mean only one list item for all the Scpoce/ActiveSpan changes?

pavolloffay replied:
Yes as it was part of one PR

tedsuo replied:
I looked though the PRs... unfortunately I don't think the Pull Requests line up cleanly, as we went through many revisions. I will try to find a middle ground.

yurishkuro commented:
should we adopt the http://keepachangelog.com/en/1.0.0/ format?

tedsuo commented:
@yurishkuro http://keepachangelog.com/en/1.0.0/ seems nice but I'd like to get a language maintainer group together to start making these decisions. There's a set of things we should standardize on, such as PR/Issue templates, release procedures, etc...

tedsuo commented:
@pavolloffay @carlosalberto @yurishkuro  I updated this PR, sorry for the delay.

- The PRs in the v0.31 were too messy to reference, as a result of numerous revisions, so I did not include them.
- We could include a link to the tracking issue, but I'm not sure how helpful that is (https://github.com/opentracing/opentracing-java/issues/189)
- Instead I simplified the line items to objects which were added and removed from the core API, and method changes on the Tracer object. Hopefully that's the right level of detail.

pavolloffay commented:
LGTM, although at least one minor yet important thing is missing. Removed `log.(Object)` methods.

pavolloffay commented:
I will merge this and open subsequent PR to add more links in there.
